app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationActivity.java

```diff
@@ -25,14 +51,123 @@ protected void onCreate(Bundle savedInstanceState) {
+        || binding.repassword.getText().toString().isEmpty()) {
+            providerRegistrationViewmodel.setRegisterError("Please enter username and password");
+            return; // early return on invalid username or password
+        }else if(!binding.password.getText().toString().equals(binding.repassword.getText().toString())){
+            providerRegistrationViewmodel.setRegisterError("Passwords do not match");
+            return;
+            }else if(binding.email.getText().toString().isEmpty() || binding.firstname.getText().toString().isEmpty()
+        || binding.lastname.getText().toString().isEmpty() || binding.home.getText().toString().isEmpty()
+        ||binding.phonenumber.getText().toString().isEmpty()){
+            providerRegistrationViewmodel.setRegisterError("Please enter your personal information");
+            return;}
+            providerRegistrationViewmodel.setRegisterError("Registering");
+            requestPermissions(new String[]{Manifest.permission.INTERNET}, REQUEST_INTERNET_REGISTER);
+            new Thread(() -> {
+                try {
+                    Thread.sleep(750);
+                } catch (InterruptedException ignored) {
 
+                }
+                runOnUiThread(() -> {
+                    Intent intent = new Intent(ProviderRegistrationActivity.this, FrontPageActivity.class);
+                    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+                    startActivity(intent);
+                });
+            }).start();
     }
+
+    /**
+     * Updates this {@link RegisterResult} to reflect the passed register result
+     *
+     * @param registerResult {@link Optional} {@link RegisterResult} to update on
+     */
+    private void updateOnRegisterResult(Optional<RegisterResult> registerResult) {
+        if(registerResult.isPresent()){
+            updateOnPresentRegisterResult(registerResult.get());
+        } else {
+            binding.registerErrorText.setVisibility(View.GONE);
+        }
+    }
+
+    /**
+     * Updates the UI based on a {@link RegisterResult}. This is called with non-null register results, meaning
+     * a registration has been attempted, and has either succeeded or failed. This is unlike updateOnRegisterResult
+     * which is called with an {@link Optional} {@link RegisterResult}, and calls this method if that
+     * {@link Optional} is present.
+     *
+     * @param result {@link RegisterResult} of login attempt
+     */
+    private void updateOnPresentRegisterResult(RegisterResult result) {
+        if (result.isSuccess()) {
+            binding.registerErrorText.setText("Register successful");
+            storeCredentials(result.getUser(), result.getToken());
+            Toast.makeText(this, "Register successful", Toast.LENGTH_SHORT);
+            startActivity(new Intent(ProviderRegistrationActivity.this, FrontPageActivity.class));
```

Comment:
Since this is registering in as Provider, I think we should go to frontpage-2 activity and not frontpage because this one is client's.